packages/next/src/server/base-server.ts

```diff
@@ -472,24 +477,21 @@ export default abstract class Server<
 
     this.enabledDirectories = this.getEnabledDirectories(dev)
 
+    const isAppPPREnabled =
+      this.enabledDirectories.app &&
+      checkIsAppPPREnabled(this.nextConfig.experimental.ppr)
+
     this.normalizers = {
       // We should normalize the pathname from the RSC prefix only in minimal
       // mode as otherwise that route is not exposed external to the server as
       // we instead only rely on the headers.
       postponed:
-        this.enabledDirectories.app &&
-        this.nextConfig.experimental.ppr &&
-        this.minimalMode
+        isAppPPREnabled && this.minimalMode
           ? new PostponedPathnameNormalizer()
           : undefined,
-      rsc:
-        this.enabledDirectories.app && this.minimalMode
-          ? new RSCPathnameNormalizer()
-          : undefined,
+      rsc: this.minimalMode ? new RSCPathnameNormalizer() : undefined,
```

Comment:
should this still be gated by `this.enabledDirectories.app`?

Reply:
I moved the check higher, so it's a part of `isAppPPREnabled`. I'll move it back as the access here for the rest all still uses it.